### PR TITLE
[Envers] HHH-17652 Fix NullPointerException when mapping audited many-to-one relationships

### PR DESCRIPTION
**Pull request by csaadaam**
fixes b384b37f395f14f5e64c6b6d4d00e0868d62fe01

https://hibernate.atlassian.net/browse/HHH-17652

### CODE REVIEW

File: hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/ToOneIdMapper.java [hunk 1]
```diff
@@ -196,6 +196,9 @@ private boolean isIgnoreNotFound(
 			EntityInfo referencedEntity,
 			Map data,
 			Object primaryKey) {
+		if (referencedEntity.isAudited()) {
+			return false;
+		}
```
Comment (mbladel) [inline hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/ToOneIdMapper.java:201]: Please follow our [code style](https://hibernate.org/community/contribute/intellij-idea/#code-style) guidelines.

```suggestion
		if ( referencedEntity.isAudited() ) {
			return false;
		}
```

**Review by mbladel — COMMENTED** [review]
@csaadaam thank you for your contribution! Sorry for taking so long, we have a long backlog of issues and we're trying to get to pending PRs.

I've left a small comment about code style, but other than that, your change looks reasonable to me. I would ask you to please include a test in your PR which ensures the NPE has been fixed by your changes and prevents us from re-introducing a regression in the future.

**Comment by mbladel** [comment]
We're also in the process or investigating a move to the Apache Software License v2. If you agree with allowing this change to be relicensed under that license, please add this text to the subject -

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

**Comment by mbladel** [comment]
@csaadaam I took the liberty of looking into this issue and found that while your solution resolved the problem with your use-case it caused errors in several envers verifying the correct functionality of audited entities using not-found relationships.

I've added a test case and proposed a fix working here: https://github.com/hibernate/hibernate-orm/pull/9453

**Comment by csaadaam** [comment]
> @csaadaam I took the liberty of looking into this issue and found that while your solution resolved the problem with your use-case it caused errors in several envers verifying the correct functionality of audited entities using not-found relationships.
> 
> I've added a test case and proposed a working fix here: #9453

@mbladel Thank you very much for looking into it and fixing it!

**Pull request closed without merging**